doc/USER_GUIDE.md

```diff
@@ -0,0 +1,132 @@
+# Installation
+
+To configure a GitHub repository to use TACOS-GHA, you need to configure a
+number of workflows in the `.github/workflows/` directory in the repository.
+Examples of these workflow files can be found [here](workflows).
+
+# Concepts
+
+- slice
+  - A collection of terraform resources sharing a single non-local terraform
+    state backend.
+- lock
+  - A standard terraform lock, thus visible to any tool using terraform.
```

Comment:
If we're gearing this towards Terraform n00bs, then we should try to explain what a lock does.

Reply:
It's good to prefer the term "terraform state lock" because most googling for "terraform lock" will talk about `.terraform.lock.hcl` which is a very different idea.

Reply:
Added that wording and an explanation.